pupil_src/shared_modules/file_methods.py

```diff
@@ -93,3 +118,182 @@ def next_export_sub_dir(root_export_dir):
+        def purge_cache(self):
+            pass
+
+    _cache_ref = [Empty()]*cache_len
+
+    #an Immutable dict for dics nested inside this dict.
+    class FrozenDict(dict):
+
+        def __setitem__(self, key, value):
+            raise NotImplementedError('Invalid operation')
+
+        def clear(self):
+            raise NotImplementedError()
+
+        def update(self, *args, **kwargs):
+            raise NotImplementedError()
+
+    def __init__(self, mapping=None,payload=None):
+        if type(mapping) is dict:
+            self._ser_data = msgpack.dumps(mapping,use_bin_type=True)
+        elif type(payload) is bytes:
+            self._ser_data = payload
+        else:
+            raise Exception("neither mapping nor payload is supplied or wrong format.")
+        self._data = None
+
+
+    def _object_hook(self,obj):
+        if type(obj) is dict:
+            return self.FrozenDict(obj)
+
+    def _deser(self):
+        if not self._data:
+            self._data = msgpack.loads(self._ser_data,raw=False,use_list=False, object_hook=self._object_hook)
+            self._cache_ref.pop(0).purge_cache()
+            self._cache_ref.append(self)
+
+    def purge_cache(self):
+        self._data = None
+
+    def __setitem__(self, key, item):
+        raise NotImplementedError()
+        self._data[key] = item
+
+    def __getitem__(self, key):
+        self._deser()
+        return self._data[key]
+
+    def __repr__(self):
+        self._deser()
+        return repr(self._data)
+
+    def __len__(self):
+        self._deser()
+        return len(self._data)
+
+    def __delitem__(self, key):
+        raise NotImplementedError()
+        self._deser()
+        del self._data[key]
+
+    def get(self,key,default):
+        try:
+            return self['key']
+        except KeyError:
+            return default
+
+    def clear(self):
+        raise NotImplementedError()
+        return self._data.clear()
+
+    def copy(self):
+        self._deser()
+        return self._data.copy()
+
+    def has_key(self, k):
+        self._deser()
+        return k in self._data
+
+    def update(self, *args, **kwargs):
+        raise NotImplementedError()
+        self._deser()
+        return self._data.update(*args, **kwargs)
+
+    def keys(self):
+        self._deser()
+        return self._data.keys()
+
+    def values(self):
+        self._deser()
+        return self._data.values()
+
+    def items(self):
+        self._deser()
+        return self._data.items()
+
+    def pop(self, *args):
+        raise NotImplementedError()
+        return self._data.pop(*args)
+
+    def __cmp__(self, dict_):
+        self._deser()
+        return self.__cmp__(self._data, dict_)
+
+    def __contains__(self, item):
+        self._deser()
+        return item in self._data
+
+    def __iter__(self):
+        self._deser()
+        return iter(self._data)
+
+    def __unicode__(self):
+        self._deser()
+        return unicode(repr(self._data))
```

Comment:
`unicode()` is not Python 3 function. I guess that the original code was meant to be written for Python 2.X. Can this function be removed?